application-admintools-default/src/main/java/com/xwiki/admintools/script/AdminToolsScriptService.java

```diff
@@ -172,13 +172,28 @@ public String getInstanceSizeSection() throws AccessDeniedException
      * @param order the order of the sort.
      * @return a {@link List} with the documents that have more than the given number of comments.
      */
-    public List<XWikiDocument> getPagesOverGivenNumberOfComments(long maxComments, Map<String, String> filters,
+    public List<DocumentReference> getPagesOverGivenNumberOfComments(long maxComments, Map<String, String> filters,
```

Comment:
This is public API and should not be changed easily, since it breaks backwards-compatibility
This is why the [Unstable](https://dev.xwiki.org/xwiki/bin/view/Community/DevelopmentPractices#H40UnstableAnnotation) annotation is used, to let devs know this is not yet stable and might suffer modifications.
So this annotation should be added here as well  (and for other public APIs you might have here)